### PR TITLE
X1034 React 18 upgrade

### PR DESCRIPTION
**Pull request by seenanair**
Upgraded
 - React 18
 - Cypress
 - Jest, jest-dom
 - React table

### CODE REVIEW

File: src/pages/Location.tsx [hunk 1]
```diff
@@ -650,12 +650,13 @@ const Location: React.FC<LocationProps> = ({ storageLocation, locationSearchPara
...
-      <PromptOnLeave
-        when={awaitingLabwares.length > 0}
-        messageHandler={awaitingStorageCheckOnExit}
-        message={'You have labwares that are not stored. Are you sure you want to leave?'}
-        onPromptLeave={onLeave}
-      />
+      {awaitingLabwares.length > 0 && (
+        <PromptOnLeave
+          when={awaitingStorageCheckOnExit}
+          message={'You have labwares that are not stored. Are you sure you want to leave?'}
```
Comment (khelwood) [inline src/pages/Location.tsx:656]: Labware is a mass noun. Can this be changed to "You have unstored labware." please?

File: src/pages/ProbeHybridisationQC.tsx [hunk 2]
```diff
@@ -392,7 +392,6 @@ export default function ProbeHybridisationQC({ comments }: SectionComments) {
               <OperationCompleteModal
                 show={submissionResult !== undefined}
                 message="Probe Hybridisation QC recorded for all labware(s)"
-                onReset={reload}
```
Comment (khelwood) [inline src/pages/ProbeHybridisationQC.tsx:395]: Why is the `onReset={reload}` removed here and several other places?

File: src/pages/Location.tsx (hunk 1 above)
Comment (seenanair) [reply src/pages/Location.tsx:656]: 👍

File: src/pages/ProbeHybridisationQC.tsx (hunk 2 above)
Comment (seenanair) [reply src/pages/ProbeHybridisationQC.tsx:395]: OperationCompleteModal class is now changed with `react-router-dom` update. Now, the reset button within '`OperationCompleteModal`' calls '`reload`' function by default  and this `onReset` prop is required only if we want to override the default action.

File: src/pages/Location.tsx [hunk 3]
```diff
@@ -653,7 +653,7 @@ const Location = () => {
       {awaitingLabwares.length > 0 && (
         <PromptOnLeave
           when={awaitingStorageCheckOnExit}
-          message={'You have labwares that are not stored. Are you sure you want to leave?'}
+          message={'You have labware that are not stored. Are you sure you want to leave?'}
```
Comment (khelwood) [inline src/pages/Location.tsx:656]: Please either "labware that is unstored" or preferably "unstored labware"

Comment (seenanair) [reply src/pages/Location.tsx:656]: Is this ok - "You have unstored labware. Are you sure you want to leave?"

Comment (khelwood) [reply src/pages/Location.tsx]: 👍

**Review by khelwood — APPROVED** [review]

**Pull request merged**